hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java

```diff
@@ -179,6 +184,61 @@ private void handleClusterShutdown(HttpPostCommand command) throws UnsupportedEn
         textCommandService.sendResponse(command);
     }
 
+    private void handleListNodes(HttpPostCommand command) throws UnsupportedEncodingException {
+        byte[] data = command.getData();
+        String[] strList = bytesToString(data).split("&");
+        String groupName = URLDecoder.decode(strList[0], "UTF-8");
+        String groupPass = URLDecoder.decode(strList[1], "UTF-8");
+        String res = "{\"status\":\"${STATUS}\" \"response\":\"${RESPONSE}\"}";
+        try {
+            Node node = textCommandService.getNode();
+            ClusterService clusterService = node.getClusterService();
+            GroupConfig groupConfig = node.getConfig().getGroupConfig();
+            if (!(groupConfig.getName().equals(groupName) && groupConfig.getPassword().equals(groupPass))) {
+                res = res.replace("${STATUS}", "forbidden");
+            } else {
+                res = res.replace("${STATUS}", "success");
+                res = res.replace("${RESPONSE}", clusterService.getMembers().toString() + "\n"
+                        + BuildInfoProvider.getBuildInfo().getVersion() + "\n"
+                        + System.getProperty("java.version"));
+                command.setResponse(HttpCommand.CONTENT_TYPE_JSON, stringToBytes(res));
+                textCommandService.sendResponse(command);
+                return;
+            }
+        } catch (Throwable throwable) {
+            res = res.replace("${STATUS}", "fail");
+        }
+        command.setResponse(HttpCommand.CONTENT_TYPE_JSON, stringToBytes(res));
+        textCommandService.sendResponse(command);
+    }
+
+    private void handleKillNode(HttpPostCommand command) throws UnsupportedEncodingException {
+        byte[] data = command.getData();
+        String[] strList = bytesToString(data).split("&");
+        String groupName = URLDecoder.decode(strList[0], "UTF-8");
+        String groupPass = URLDecoder.decode(strList[1], "UTF-8");
+        String res = "{\"status\":\"${STATUS}\"}";
+        try {
+            Node node = textCommandService.getNode();
+            ClusterService clusterService = node.getClusterService();
+            GroupConfig groupConfig = node.getConfig().getGroupConfig();
+            if (!(groupConfig.getName().equals(groupName) && groupConfig.getPassword().equals(groupPass))) {
+                res = res.replace("${STATUS}", "forbidden");
+            } else {
+                res = res.replace("${STATUS}", "success");
+                command.setResponse(HttpCommand.CONTENT_TYPE_JSON, stringToBytes(res));
+                textCommandService.sendResponse(command);
```

Comment:
Can you please remove following two lines + `return` keyword at the end of the else block?

```
command.setResponse(HttpCommand.CONTENT_TYPE_JSON, stringToBytes(res));
                textCommandService.sendResponse(command);
```